livy/livy.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -x -e
+
+function update_apt_get() {
+  for ((i = 0; i < 10; i++)); do
+    if apt-get update; then
+      return 0
+    fi
+    sleep 5
+  done
+  return 1
+}
+
+# Only run this initialization action on the master node.
+ROLE=$(/usr/share/google/get_metadata_value attributes/dataproc-role)
+if [[ "${ROLE}" != 'Master' ]]; then
+  exit 0
+fi
+
+APACHE_MIRROR="https://www.apache.org/dyn/mirrors/mirrors.cgi?action=download&filename"
+
+LIVY_DIR="/usr/lib/livy"
+LIVY_CONF="/usr/lib/livy/livy-0.5.0-incubating-bin/conf"
+
+# Install Livy.
+wget -O "livy.zip" "${APACHE_MIRROR}=incubator/livy/0.5.0-incubating/livy-0.5.0-incubating-bin.zip"
+unzip "livy.zip" -d "${LIVY_DIR}"
+
+# Generate livy environment file.
+cat << EOF > "${LIVY_CONF}/livy-env.sh"
+export SPARK_HOME=/usr/lib/spark
+export SPARK_CONF_DIR=/etc/spark/conf
+export HADOOP_CONF_DIR=/etc/hadoop/conf
+export LIVY_LOG_DIR=/var/log/livy
```

Comment:
You might want to use `install -d` and `chown` to create the log dir with the correct permissions.